examples/jsm/nodes/utils/RotateNode.js

```diff
@@ -0,0 +1,46 @@
+import TempNode from '../core/TempNode.js';
+import { addNodeClass } from '../core/Node.js';
+import {
+	addNodeElement,
+	nodeProxy,
+	vec4,
+	mat4,
+} from '../shadernode/ShaderNode.js';
+import { cos, sin } from '../math/MathNode.js';
+
+class RotateNode extends TempNode {
+
+	constructor( positionNode, rotationNode ) {
+
+		super( 'vec3' );
+
+		this.positionNode = positionNode;
+		this.rotationNode = rotationNode;
+
+	}
+
+	setup() {
+
+
+
+		const { rotationNode, positionNode } = this;
+
+		const rotation = rotationNode;
+		const rotationXMatrix = mat4( vec4( 1.0, 0.0, 0.0, 0.0 ), vec4( 0.0, cos( rotation.x ), sin( rotation.x ).negate(), 0.0 ), vec4( 0.0, sin( rotation.x ), cos( rotation.x ), 0.0 ), vec4( 0.0, 0.0, 0.0, 1.0 ) );
```

Comment:
IIRC (not sure, if not, then we should definitely add this) something like `mat4( a1, a2, ..., a16 )` should work even without using `vec4`.